test/features/telemetryingest/telemetryingest.go

```diff
@@ -0,0 +1,419 @@
+
+	secretConfig := tenant.GetSingleTenantSecret(t)
+
+	optionsZipkin := []componentDynakube.Option{
+		componentDynakube.WithApiUrl(secretConfig.ApiUrl),
+		componentDynakube.WithTelemetryIngestEnabled(true, "zipkin"),
+	}
+
+	testDynakubeZipkin := *componentDynakube.New(optionsZipkin...)
+
+	componentDynakube.Install(builder, helpers.LevelAssess, &secretConfig, testDynakubeZipkin)
+
+	builder.Assess("otel collector started", statefulset.WaitFor(testDynakubeZipkin.OtelCollectorStatefulsetName(), testDynakubeZipkin.Namespace))
+	builder.Assess("otel collector config created", checkOtelCollectorConfig(&testDynakubeZipkin))
+	builder.Assess("otel collector service created", checkOtelCollectorService(&testDynakubeZipkin))
+
+	var zipkinConfigResourceVersion string
+	builder.Assess("otel collector configuration timestamp", getOtelCollectorConfigResourceVersion(&testDynakubeZipkin, &zipkinConfigResourceVersion))
+
+	var zipkinPodStartTs time.Time
+	builder.Assess("otel collector pod creation timestamp", getOtelCollectorPodTimestamp(&testDynakubeZipkin, &zipkinPodStartTs))
+
+	optionsJaeger := []componentDynakube.Option{
+		componentDynakube.WithApiUrl(secretConfig.ApiUrl),
+		componentDynakube.WithTelemetryIngestEnabled(true, "jaeger"),
+	}
+
+	testDynakubeJaeger := *componentDynakube.New(optionsJaeger...)
+	componentDynakube.Update(builder, helpers.LevelAssess, testDynakubeJaeger)
+
+	builder.Assess("otel collector started", statefulset.WaitFor(testDynakubeJaeger.OtelCollectorStatefulsetName(), testDynakubeJaeger.Namespace))
+	builder.Assess("otel collector config created", checkOtelCollectorConfig(&testDynakubeJaeger))
+	builder.Assess("otel collector service created", checkOtelCollectorService(&testDynakubeJaeger))
+
+	var jaegerConfigResourceVersion string
+	builder.Assess("otel collector configuration timestamp", getOtelCollectorConfigResourceVersion(&testDynakubeJaeger, &jaegerConfigResourceVersion))
+	builder.Assess("otel collector configuration updated", func(ctx context.Context, t *testing.T, config *envconf.Config) context.Context {
+		assert.NotEqual(t, jaegerConfigResourceVersion, zipkinConfigResourceVersion)
+
+		return ctx
+	})
+
+	var jaegerPodStartTs time.Time
+	builder.Assess("otel collector pod creation timestamp", getOtelCollectorPodTimestamp(&testDynakubeJaeger, &jaegerPodStartTs))
+	builder.Assess("otel collector pod restarted", func(ctx context.Context, t *testing.T, config *envconf.Config) context.Context {
+		assert.Greater(t, jaegerPodStartTs, zipkinPodStartTs)
+
+		return ctx
+	})
+
+	componentDynakube.Delete(builder, helpers.LevelTeardown, testDynakubeJaeger)
+
+	builder.WithTeardown("deleted tenant secret", tenant.DeleteTenantSecret(testDynakubeJaeger.Name, testDynakubeJaeger.Namespace))
+
+	return builder.Feature()
+}
+
+// Make sure the Otel collector StS and config ConfigMap are removed when telemetryIngest gets disabled
+func OtelCollectorCleanup(t *testing.T) features.Feature {
```

Comment:
does this have to be a separate test?

couldn't it be just the last step in one of the other OtelCollector tests?